src/main/resources/templates/fragments/usersFragment.html

```diff
@@ -0,0 +1,55 @@
+<!DOCTYPE html>
+<html lang="en"
+      xmlns:th="http://www.thymeleaf.org"
+      xmlns:layout="http://www.ultraq.net.nz/thymeleaf/layout"
+      layout:decorate="~{layout/main}">
+<body>
+<!-- in usersFragment.html -->
+
+<div th:fragment="combinedFragment" id="replace-me" hx-swap="outerHTML">
+    <div class="flex flex-col items-center justify-center h-screen gap-10">
+
+        <div class="join">
+            <div>
+                <div>
+                    <input id="user-search" class="input input-bordered join-item" placeholder="Search"/>
+                </div>
+            </div>
+            <select id="user-select" class="select select-bordered join-item">
+                <option disabled selected>Select existing user</option>
+                <option th:each="user : ${users}" th:value="${user.id}"
+                        th:text="|${user.firstName}  ${user.lastName} (${user.email})|">User
+                </option>
+            </select>
+            <div class="indicator">
+                <button class="btn join-item">Select</button>
+            </div>
+        </div>
+
+        <div class="divider">OR</div>
+
+        <a id="create-new-user-button" hx:get="@{/users/create}"
+           hx-swap="outerHTML" hx-trigger="click"
+           hx-target="#replace-me"
+           class="btn btn-outline btn-secondary w-full max-w-md"
+           role="button">Create new user</a>
```

Comment:
I like the HTMX, but I do wonder if it wouldn't generally be better to just have a new page for this. I think the general idea of HTMX is to do HTML swaps for small parts of the view. But if we're just going to replace all the HTML, then we just need a link, not HTMX. Still, I get that this is an experiment so no worries!

Reply:
I'd rather to new pages for everything, but one of my goals here is to see if I can a SPA. Is that a misuse of htmx?